doc/pages/expansions.asciidoc

```diff
@@ -1,199 +1,294 @@
 = Expansions
 
-== Strings
+While parsing a command (see <<command-parsing#,`:doc command-parsing`>>),
+Kakoune recognises certain patterns and will replace them with their
+associated value before executing the command. These patterns are called
+expansions.
 
-\'strings'::
-    uninterpreted strings, use a backslash (\') to escape the separator
+Every expansion consists of a `%`, followed by the expansion _type_ (one
+or more alphabetic characters), a nestable punctuation character (`(`, `[`,
+`{`, or `<`), and then all the text up to and including its matching opposite
+(`)`, `]`, `}`, or `>`). It doesn't matter which character is used, but
+`{}` are most common. For example, the command `echo %val{session}` will
+echo the ID of the current Kakoune session.
 
-"strings"::
-    expanded strings, % strings (c.f. next section) contained are expended,
-    use a backslash (\\%) to escape the separator
+Expansions are processed when unquoted and anywhere inside double-quoted or
+%-strings, but not inside unquoted words, inside single-quoted strings, or
+inside other expansions. So:
 
-%\{strings\}::
-    these strings are very useful when entering commands
+* `echo %val{session}` echoes the current session ID
 
-    * the '{' and '}' delimiters are configurable, any non alphanumeric
-      character can be used
+* `echo x%val{session}x` echoes the literal text `x%val{session}x`
 
-----------------------------------------------------------
-e.g. %[string], %<string>, %(string), %~string~, %!string!
-----------------------------------------------------------
+* `echo '%val{session}'` echoes the literal text `%val{session}`
 
-    * if the character following '%' is one of '{[(<', then the closing
-      one is the matching '}])>' and the delimiters are not escapable but
-      are nestable
+* `echo "x%val{session}x"` echoes the current session ID, surrounded by `x`
 
------------------------------------------------------------
-e.g. %{ roger {}; } is a valid string, %{ marcel \} as well
------------------------------------------------------------
+* `echo %{x%val{session}x}"` echoes the current session ID, surrounded by `x`
 
-== Typed expansions
+* `echo %sh{ echo %val{session} }"` echoes the literal text `%val{session}`
 
-%\{strings\} can have an expansion type between the *%* and the opening
-character. They will be written *%<type>\{<content>\}*. They will be
-expanded according to the given *<type>* using *<content>* as its
-parameter:
+When processing an expansion, Kakoune ensures the result is properly quoted so
+that any special characters do not interfere with the rest of the command. If
+an expansion expands to multiple values, each value is individually quoted.
```

Comment:
I am not sure `quoted` is correct here, expansions can expand to multiple words, but they are not reparsed at all.